src/main/java/io/kamax/matrix/client/MatrixHttpRoom.java

```diff
@@ -268,4 +274,120 @@ public _MatrixRoomMessageChunk getMessages(_MatrixRoomMessageChunkOptions option
         return new MatrixRoomMessageChunk(body.getStart(), body.getEnd(),
                 body.getChunk().stream().map(MatrixJsonPersistentEvent::new).collect(Collectors.toList()));
     }
+
+    @Override
+    public List<RoomTag> getUserTags() {
+        return getAllTags().stream().filter(tag -> "u".equals(tag.getNamespace())).collect(Collectors.toList());
+    }
+
+    private List<RoomTag> getAllTags() {
+        List<RoomTag> tags = new ArrayList<>();
+
+        URIBuilder builder = getClientPathBuilder("/user/{userId}/rooms/{roomId}/tags");
+
+        String body = execute(new HttpGet(getWithAccessToken(builder)));
+
+        if (StringUtils.isNotEmpty(body)) {
+            JsonObject jsonTags = jsonParser.parse(body).getAsJsonObject().get("tags").getAsJsonObject();
+            tags = jsonTags.entrySet().stream().filter(e -> e.getValue().isJsonObject()).map(entry -> {
+                String completeName = entry.getKey();
+                String name = "";
+                String namespace = "";
+                if (completeName.startsWith("m.")) {
+                    name = completeName.substring(2);
+                    namespace = "m";
+                } else if (completeName.startsWith("u.")) {
+                    name = completeName.substring(2);
+                    namespace = "u";
+                } else {
+                    name = completeName;
+                }
+                JsonElement jsonOrder = entry.getValue().getAsJsonObject().get("order");
+                Double order = null;
+                if (jsonOrder != null) {
+                    order = jsonOrder.getAsDouble();
+                }
+
+                return new RoomTag(namespace, name, order, roomId);
+            }).collect(Collectors.toList());
+        }
+
+        return tags;
+    }
+
+    @Override
+    public void addUserTag(String tag) {
+        addTag("u." + tag, 1);
```

Comment:
Why `1` as default and not `null`?

Reply:
To my understanding, `null` is not allowed by the spec: https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#id451

It states, that the order has to be in the range of `[0,1]`.

Reply:
In case of the JSON library, `null` would mean the key is not added to the JSON string, which would be fine since `order` is not marked as mandatory (and so, optional)